include/pipettes/core/tasks/i2c_task.hpp

```diff
@@ -33,18 +33,46 @@ class I2CMessageHandler {
     void visit(ReadFromI2C &m) {
         i2c_device.central_receive(m.buffer.data(), m.buffer.size(), m.address,
                                    TIMEOUT);
-        m.client_callback(m.buffer);
+        m.handle_buffer(m.buffer);
+        m.client_callback();
     }
 
-    void visit(PollReadFromI2C &m) {
+    void visit(SingleRegisterPollReadFromI2C &m) {
```

Comment:
It looks like processing `SingleRegisterPollReadFromI2C` and `MultiRegisterPollReadFromI2C`  messages will keep this task busy for the duration of the poll. 

Is there a concern that blocking all other I2C communication during the poll can be a problem?

Should the polling be happening from a different task?

Reply:
I wanted it to block i2c communication because you can't read/write from two things at the same time on the same bus which is why I had it block in the i2c task.